### PR TITLE
zypperpkg fix vendor change logic

### PR DESCRIPTION
**Pull request by mseidl**
### What does this PR do?
Fixes the vendor change behavior. Adds "allowvendorchange" option, and improves the check for supported zypper versions.  This is an expansion to the old PR which I will get close.  https://github.com/saltstack/salt/pull/58560

### What issues does this PR fix or reference?

Fixes: No issue

### Previous Behavior

What happens with novendorchange is: the option of "--no-allow-change" is visible or not. The issue is, in zypper vendor changes are not allowed by default. So, the the behavior between "zypper dup --no-allow-vendor-change" and "zypper dup" is nothing...

### New Behavior

Will add the "--allow-vendor-change option if supported.  The allowvendorchange option is better fit for how zypper works. 

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [ ] Changelog - https://docs.saltstack.com/en/master/topics/development/changelog.html
- [ ] Tests written/updated

### Commits signed with GPG?
No.

Please review [Salt's Contributing Guide](https://docs.saltstack.com/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: salt/modules/zypperpkg.py [hunk 1]
```diff
@@ -1744,7 +1807,11 @@ def upgrade(
         Specify a list of package repositories to upgrade from. Default: None
 
     novendorchange
-        If set to True, no allow vendor changes. Default: False
+        DEPRECATED(use allowvendorchange): If set to True, do not allow vendor changes. Default: True
```
Comment (waynew) [inline salt/modules/zypperpkg.py:1810]: **question** Do we intend to actually remove this? If we do then we should add a `warn_until` if `novendorchange` is different. It actually might be a good idea to set the default to None in the function call, so...

```
def upgrade(..., novendorchange=None,...):
    if novendorchange is not None:
        warn_until(..., "novendorchange is deprecated and will be removed after the XXX release.")
    else:
       novendorchange = not allowvendorchange
```

See https://archive.repo.saltproject.io/en/2015.5/topics/development/deprecations.html for more info.

**Review by waynew — CHANGES_REQUESTED** [review]
Hey, sorry for the delay in review - this appears to have slipped through the cracks.

Only one question, looks great otherwise!

File: salt/modules/zypperpkg.py (hunk 1 above)
Comment (mseidl) [reply salt/modules/zypperpkg.py:1810]: Well, I'd recommend getting rid of `novenderchange`.   It was the wrong option from the beginning.

Comment (waynew) [reply salt/modules/zypperpkg.py:1810]: If it's completely broken (i.e. it's a bug) then that's fine, but even if it's a nonsense functionality we still don't want to just break things for people who may be relying on it. Or just using it.

Comment (mseidl) [reply salt/modules/zypperpkg.py:1810]: It never worked from the beginning.  But I'm not saying it needs to be removed right this second.

Comment (waynew) [reply salt/modules/zypperpkg.py:1810]: My suggestion then, is to go ahead and use `warn_until` - can even mention something in the warning about how `novendorchange` never worked right or whatever.

**Comment by waynew** [comment]
Looks like tests were failing but they may have been unrelated. I went ahead and merged in the latest changes so we'll see how this goes :+1:

**Comment by mseidl** [comment]
> Looks like tests were failing but they may have been unrelated. I went ahead and merged in the latest changes so we'll see how this goes +1

I had all the tests working locally. But @meaksh  pointed out that there was a test here failing  that wasn't my fault.

**Comment by dwoz** [comment]
Closing this due to inactivity. Anyone should feel free to re-open it if they want to see it through to the end in one release cycle.

**Pull request closed without merging**